qa/smoke-test-http/src/javaRestTest/java/org/elasticsearch/http/BlockedSearcherRestCancellationTestCase.java

```diff
@@ -149,12 +155,36 @@ private static class SearcherBlockingEngine extends ReadOnlyEngine {
 
         @Override
         public Searcher acquireSearcher(String source, SearcherScope scope, Function<Searcher, Searcher> wrapper) throws EngineException {
+            if (testCaseLogger.isDebugEnabled()) {
+                testCaseLogger.debug(
+                    Strings.format(
+                        "in acquireSearcher for shard [%s] on thread [%s], availablePermits=%d",
+                        config().getShardId(),
+                        Thread.currentThread().getName(),
+                        searcherBlock.availablePermits()
+                    ),
+                    new ElasticsearchException("stack trace")
+                );
+            }
+
             try {
                 searcherBlock.acquire();
             } catch (InterruptedException e) {
                 throw new AssertionError(e);
             }
             searcherBlock.release();
+
+            if (testCaseLogger.isDebugEnabled()) {
```

Comment:
This check is just to ascertain whether to do work involved in preparing the DEBUG log message, right? Do we care about that performance in testing infrastructure? Or is this simply following good coding practice?

Reply:
Test performance is less important than production performance but not totally unimportant. Probably doesn't matter here too much but a bare `logger.debug()` with expensive params should just look wrong regardless of context.